orm.go

```diff
@@ -59,14 +59,21 @@ func NewSchema() *Schema {
 }
 
 // Index returns an index with a name.
-func (s *Schema) Index(name string) (*Index, error) {
+func (s *Schema) Index(name string, options ...interface{}) (*Index, error) {
```

Comment:
Would it make sense to make this function to make `options` to take `...IndexOption` and remove the support for `*IndexOptions` ? I will modify `index.Field` to take `...FieldOption`.